commit 5dda1abc32b7b1eb59901a5592569e21d88c6ad2
Author: Nicholas Marriott <nicm@openbsd.org>
Date:   Sun Apr 21 21:32:00 2013 +0000

    Don't let server_client_check_focus use a dead bufferevent, from Romain
    Francoise.

diff --git a/server-client.c b/server-client.c
index 6cacefa..3b7b988 100644
--- a/server-client.c
+++ b/server-client.c
@@ -514,8 +514,10 @@ server_client_loop(void)
 
 		w->flags &= ~WINDOW_REDRAW;
 		TAILQ_FOREACH(wp, &w->panes, entry) {
-			server_client_check_focus(wp);
-			server_client_check_resize(wp);
+			if (wp->fd != -1) {
+				server_client_check_focus(wp);
+				server_client_check_resize(wp);
+			}
 			wp->flags &= ~PANE_REDRAW;
 		}
 	}
@@ -527,7 +529,7 @@ server_client_check_resize(struct window_pane *wp)
 {
 	struct winsize	ws;
 
-	if (wp->fd == -1 || !(wp->flags & PANE_RESIZE))
+	if (!(wp->flags & PANE_RESIZE))
 		return;
 
 	memset(&ws, 0, sizeof ws);
